src/inputs.ts

```diff
@@ -0,0 +1,119 @@
+import * as core from '@actions/core'
+import type * as github from '@actions/github'
+import { pullRequestSchema } from './types'
+import type { PullRequest, Octokit } from './types'
+import type { Config } from './config'
+
+export const inputs = {
+  getToken() {
+    return core.getInput('github-token', { required: true, trimWhitespace: true })
+  },
+
+  getBodyRegex() {
+    const string = core.getInput('body-regex', { required: true })
+    return new RegExp(string)
+  },
+
+  getBodyTemplate() {
+    const template = core
+      .getMultilineInput('body-template', { required: true })
+      .join('\n')
+
+    if (!/{{ *stack *}}/.exec(template)) {
+      core.setFailed('Input `body-template` is missing `{{ stack }}`')
+      process.exit()
+    }
+
+    return template
+  },
+
+  async getMainBranch(
+    octokit: Octokit,
+    config: Config,
+    context: typeof github.context
+  ): Promise<string> {
+    const mainBranch: string | undefined = config.branches?.main
+
+    if (mainBranch) {
+      core.info('Main branch found in config.')
+      return mainBranch
+    }
+
+    core.info(
+      'Main branch not found in config. Guessing main branch from repository default branch.'
+    )
+
+    const {
+      data: { default_branch: defaultBranch },
+    } = await octokit.rest.repos.get({
+      ...context.repo,
+    })
+
+    return defaultBranch
+  },
+
+  async getPerennialBranches(config: Config): Promise<string[]> {
+    let perennialBranches: string[] = config.branches?.perennials ?? []
+
+    if (perennialBranches.length > 0) {
+      core.info('Perennial branches found in config.')
+      return perennialBranches
+    }
+
+    core.info('Perennial branches not present in config. Using action inputs instead.')
+
+    perennialBranches = core.getMultilineInput('perennial-branches', {
+      required: false,
+      trimWhitespace: true,
+    })
+
+    if (perennialBranches.length === 0) {
+      core.warning(
+        'No perennial branches found in config or inputs. If this is a mistake, ensure `.git-branches.toml` is configured correctly when using Git Town v12+, or manually configure the `perennial-branches` input.'
+      )
+    }
+
+    return perennialBranches
+  },
+
+  getCurrentPullRequest(context: typeof github.context) {
+    try {
+      const pullRequest:
+        | {
+            number: number
+            base?: { ref?: string }
+            head?: { ref?: string }
+          }
```

Comment:
Extract this type into types.ts?

Reply:
The type is purely for working with `base` and `head`, which are both `any` due to `context.payload.pull_request` having the following type:

```ts
type WebhookPayload = {
  pull_request: {
    [key: string]: any;
    number: number;
    html_url?: string | undefined;
    body?: string | undefined;
  } | undefined
}
```

Since `inputs.ts` abstracts this away, I don't believe there's much value in extracting this type out, since (at least I hope) it wouldn't be used elsewhere.

Reply:
Cool, thanks for the background!